cmake/onnxruntime_optimizer.cmake

```diff
@@ -57,6 +57,8 @@ else()
     "${ONNXRUNTIME_INCLUDE_DIR}/core/optimizer/*.h"
     "${ONNXRUNTIME_ROOT}/core/optimizer/*.h"
     "${ONNXRUNTIME_ROOT}/core/optimizer/*.cc"
+    "${ONNXRUNTIME_ROOT}/core/optimizer/compute_optimizer/*.h"
```

Comment:
All the code in these files and in the test files is wrapped in ENABLE_TRAINING so why add these files here? You can add them within the if (onnxruntime_ENABLE_TRAINING) condition

Reply:
Because the optimizer is applicable for inferencing. I intent to make it easier to enable it for inference later without changing file location and cmake macros.